bellman/src/lib.rs

```diff
@@ -356,6 +353,30 @@ impl fmt::Display for SynthesisError {
     }
 }
 
+/// An error during verification.
+#[derive(Debug)]
+pub enum VerificationError {
+    /// Verification was attempted with a malformed verifying key.
+    MalformedVerifyingKey,
+    /// Proof verification failed.
+    Failed,
```

Comment:
nit: maybe `ProofVerificationFailed` - since any VerificationError represents a failure.

Reply:
Calling it `VerificationError::ProofVerificationFailed` seems sort of redundant, maybe `VerificationError::InvalidProof` (which fits with @str4d's suggestion for `VerificationError::InvalidVerifyingKey` for the other variant)?

Reply:
Updated in a154154